src/page/categories/Categories.jsx

```diff
@@ -0,0 +1,164 @@
+import { useState, useEffect } from 'react';
+import {
+  Container, Typography, Select, MenuItem, CircularProgress, List, ListItem, ListItemText, Button, RadioGroup, FormControlLabel, Radio
+} from '@mui/material';
+
+const Categories = () => {
+  const [categories, setCategories] = useState([]);
+  const [selectedCategory, setSelectedCategory] = useState('');
+  const [questions, setQuestions] = useState([]);
+  const [quizQuestions, setQuizQuestions] = useState([]); 
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+  const [userAnswers, setUserAnswers] = useState({});
+  const [showQuiz, setShowQuiz] = useState(false); 
+  const [score, setScore] = useState(null);  
+
+  useEffect(() => {
+    fetch('https://opentdb.com/api_category.php')
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error('Network response was not ok');
+        }
+        return response.json();
+      })
+      .then((data) => {
+        setCategories(data.trivia_categories);
+      })
+      .catch((error) => {
+        setError('Error fetching categories');
+        console.error('Error fetching categories:', error);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (selectedCategory) {
+      setLoading(true);
+      fetch(`https://opentdb.com/api.php?amount=10&type=multiple&category=${selectedCategory}`)
```

Comment:
In another file I see you are using Async/await approach and here you are using promis/then please **standardize** your code syntax.